ftrack_sftp_accessor/sftp.py

```diff
@@ -0,0 +1,258 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2014-2022 ftrack
+
+import os
+import logging
+import paramiko
+
+from ftrack_api.accessor.base import Accessor
+from ftrack_api.data import FileWrapper
+from ftrack_api.exception import (
+    AccessorOperationFailedError,
+    AccessorUnsupportedOperationError,
+    AccessorResourceInvalidError,
+    AccessorResourceNotFoundError,
+    AccessorContainerNotEmptyError,
+    AccessorParentResourceNotFoundError,
+)
+
+
+class SFTPAccessor(Accessor):
+    """Provide SFTP location access."""
+
+    def __init__(self, host, username, port=22, password=None):
```

Comment:
@iwootten 
how about providing a **prefix**  argument to define a root subfolder to upload to  ?
This would allow to have multiple accessor working on the same sftp and upload into multiple subfolders of it extending its usage. 

this would be similar on how the [DiskAccessor](https://bitbucket.org/ftrack/ftrack-python-api/src/8832bf1ab619ce0f4a7ca3f841dc5e07ab39aca8/source/ftrack_api/accessor/disk.py#lines-24) works.

Reply:
Sounds good, maybe this can be submitted as a subsequent PR later to avoid holding this one up.

Reply:
doesn't worth a new pr. let's have it as part of this initial commit, so matches the other accessor behaviour.